### ISSUE TITLE
[Checkbox]: improvements

### ISSUE THREAD

**Issue by ly-tempel-bitweb** [TEDI-Design-System/angular#417]
## Problem
1. As Checkbox label is clickable, cursor should be `cursor: pointer;`
2. Focus ring color is incorrect, should be #337BB5, `var(--tedi-primary-500)`

<img width="944" height="313" alt="Image" src="https://github.com/user-attachments/assets/3ddaaa04-2b4e-494b-95e6-decf903fa131" />

3. Add focus state under States story.

4. Error state should have a red border on hover.<img width="268" height="65" alt="Image" src="https://github.com/user-attachments/assets/1792b272-e0cc-4f67-bdcf-ccd4c5b0015a" />

**Comment by mart-sessman** [comment]
some issues were already addressed in #412, fixed others in that branch also as radio component had similar issues
